### PR TITLE
Disable selection of students who already have a mentor

### PR DESCRIPTION
**Pull request by Pulya10c**
**🟢 Add `deploy` label if you want to deploy this Pull Request to staging environment**

#### 🧑‍⚖️ Pull Request Naming Convention

- Title should follow [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary)
- Do not put issue id in title
- Do not put WIP in title. Use [Draft PR](https://github.blog/2019-02-14-introducing-draft-pull-requests/) functionality
- Consider to add `area:*` label(s)

* [x] I followed naming convention rules

---

#### 🤔 This is a ...

- [ ] New feature
- [x] Bug fix
- [ ] Performance optimization
- [ ] Refactoring
- [ ] Test Case
- [ ] Documentation update
- [ ] Other

#### 🔗 Related issue link

https://github.com/rolling-scopes/rsschool-app/issues/1494

#### 💡 Background and solution

_Describe the big picture of your changes here_

#### ☑️ Self Check before Merge

⚠️ Please check all items below before review. ⚠️

- [ ] Database migration is added or not needed
- [ ] Documentation is updated/provided or not needed
- [ ] Changes are tested locally

### CODE REVIEW

File: client/src/components/StudentSearch.tsx [hunk 1]
```diff
@@ -4,6 +4,7 @@ import { CourseService } from 'services/course';
 
 type Props = UserProps & {
   courseId: number;
+  hideWithoutMentor?: boolean;
```
Comment (aaliakseyenka) [inline client/src/components/StudentSearch.tsx:7]: kind of confusing: title says we should hide students with mentor, although your prop says opposite. Am I missing smth?

File: client/src/pages/course/mentor/confirm.tsx [hunk 2]
```diff
@@ -178,3 +174,18 @@ function renderForm(form: FormInstance, mentorData: any, handleSubmit: (values:
     </>
   );
 }
+
+const MentorConfirmPage: any = withSession(MentorConfirm);
+
+MentorConfirmPage.getInitialProps = async (context: NextPageContext) => {
```
[the remark this answers is not in the document]
Comment (apalchys) [reply client/src/pages/course/mentor/confirm.tsx:180]: yes, please use getServerSideProps. You can find example in the repo

File: client/src/pages/course/mentor/confirm.tsx [hunk 3]
```diff
@@ -178,3 +174,18 @@ function renderForm(form: FormInstance, mentorData: any, handleSubmit: (values:
     </>
   );
 }
+
+const MentorConfirmPage: any = withSession(MentorConfirm);
```
[the remark this answers is not in the document]
Comment (apalchys) [reply client/src/pages/course/mentor/confirm.tsx:178]: if you use `getServerSideProps`, you don't need any

File: client/src/components/UserSearch.tsx [hunk 4]
```diff
@@ -25,7 +26,8 @@ export function UserSearch(props: UserProps) {
   const handleSearch = async (value: string) => {
     if (value) {
       const data = await searchFn(value);
-      setData(data);
+      const persons = hideWithoutMentor ? data.filter(person => (person as SearchStudent)?.mentor == null) : data;
```
[the remark this answers is not in the document]
Comment (Pulya10c) [reply client/src/components/UserSearch.tsx]: Fixed, thanks.

File: client/src/pages/course/mentor/confirm.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (Pulya10c) [reply client/src/pages/course/mentor/confirm.tsx]: I implemented it in a different way. Thank you.

File: client/src/pages/course/mentor/confirm.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (Pulya10c) [reply client/src/pages/course/mentor/confirm.tsx]: As I wrote above, I had a call with Artem and he advised another way. Thank you.

**Review by apalchys — APPROVED** [review]

File: server/src/routes/course/students.ts [hunk 5]
```diff
@@ -46,9 +46,11 @@ export const getStudentsWithDetails = (_: ILogger) => async (ctx: Router.RouterC
 
 export const searchStudent = (_: ILogger) => async (ctx: Router.RouterContext) => {
   const { courseId, searchText } = ctx.params;
+  const { onlyStudentsWithoutMentorShown } = ctx.query;
 
   const repository = getCustomRepository(StudentRepository);
-  const result = await repository.search(Number(courseId), searchText);
+  const result = await repository.search(Number(courseId), searchText, onlyStudentsWithoutMentorShown === 'true');
+  // const result = hideStudentWithMentor === 'true' ? students.filter(student => student.mentor == null) : students;
```
Comment (aaliakseyenka) [inline server/src/routes/course/students.ts:53]: remove comment?

Comment (Pulya10c) [reply server/src/routes/course/students.ts]: fixed, tanks.

**Pull request merged**